Source/Core/DolphinQt2/GameList/GameTracker.cpp

```diff
@@ -69,26 +71,51 @@ void GameTracker::RemoveDirectory(const QString& dir)
 void GameTracker::UpdateDirectory(const QString& dir)
 {
   QDirIterator it(dir, game_filters, QDir::NoFilter, QDirIterator::Subdirectories);
+  QSet<QString> missing_files;
+
+  for (auto const& key : m_tracked_files.keys())
+  {
+    if (m_tracked_files[key].contains(dir))
+      missing_files.insert(key);
+  }
+
   while (it.hasNext())
   {
     QString path = QFileInfo(it.next()).canonicalFilePath();
+
     if (m_tracked_files.contains(path))
     {
-      m_tracked_files[path]++;
+      if (!m_tracked_files[path].contains(dir))
```

Comment:
`m_tracked_files[path]` should be stored to reference, otherwise you're doing a map lookup twice.